src/m365/commands/login.ts

```diff
@@ -239,7 +243,9 @@ class LoginCommand extends Command {
       }
     }
     finally {
-      logout();
+      if (!args.options.ensure || (args.options.ensure && !auth.service.connected)) {
```

Comment:
This doesn't match the spec at https://github.com/pnp/cli-microsoft365/issues/5217#issuecomment-1636690764, where we stated that we also want to compare additional values like tenant ID, account, etc, if specified. Let's extend it.